tests/protocols/test_http.py

```diff
@@ -153,31 +157,34 @@ def add_done_callback(self, callback):
         pass
 
 
-def get_connected_protocol(app, protocol_cls, **kwargs):
-    loop = MockLoop()
+@contextlib.contextmanager
+def get_connected_protocol(app, protocol_cls, event_loop, **kwargs):
+    loop = MockLoop(event_loop)
     transport = MockTransport()
     config = Config(app=app, **kwargs)
     server_state = ServerState()
     protocol = protocol_cls(config=config, server_state=server_state, _loop=loop)
     protocol.connection_made(transport)
-    return protocol
+    yield protocol
+    protocol.loop.close()
 
 
+@pytest.mark.asyncio
```

Comment:
Do we really need this marker, or can we have the mock setup the event loop by itself? Might help simplify things. I also find it surprising to use this marker on a sync test function (usually it is used to invoke async test functions).

Reply:
ok I thought that it was necessary because I'm using their (pytest-asyncio) `event_loop` fixture, but it is not, so I removed it.

To be complete you can use the marker on a sync test, pytest_asyncio will wrap it, but that's effectively unnecessary in our case